Source/objects.cpp

```diff
@@ -4240,73 +4240,70 @@ void OperateObject(int pnum, int i, BOOL TeleFlag)
 
 void SyncOpL1Door(int pnum, int cmd, int i)
 {
-	signed int v3;    // eax
-	ObjectStruct *v4; // esi
+	BOOL do_sync;
 
-	if (pnum != myplr) {
-		v3 = 0;
-		if (cmd == 43) {
-			if (object[i]._oVar4)
-				return;
-			v3 = 1;
-		}
-		if (cmd == 44 && object[i]._oVar4 == 1)
-			v3 = 1;
-		if (v3) {
-			v4 = &object[i];
-			if (v4->_otype == 1)
-				OperateL1LDoor(-1, i, 0);
-			if (v4->_otype == OBJ_L1RDOOR)
-				OperateL1RDoor(-1, i, 0);
-		}
+	if (pnum == myplr)
+		return;
+
+	do_sync = FALSE;
+	if (cmd == 43) {
+		if (object[i]._oVar4 != 0)
+			return;
+		do_sync = TRUE;
+	}
+	if (cmd == 44 && object[i]._oVar4 == 1)

```

Comment:
I think we have an enum for `cmd`. Try using this one (from `enums.h`):

```c
typedef enum _cmd_id {
	CMD_STAND             = 0,
	CMD_WALKXY            = 1,
	CMD_ACK_PLRINFO       = 2,
	CMD_ADDSTR            = 3,
	CMD_ADDMAG            = 4,
	CMD_ADDDEX            = 5,
	CMD_ADDVIT            = 6,
	CMD_SBSPELL           = 7,
	CMD_GETITEM           = 8,
	CMD_AGETITEM          = 9,
	CMD_PUTITEM           = 10,
	CMD_RESPAWNITEM       = 11,
	CMD_ATTACKXY          = 12,
	CMD_RATTACKXY         = 13,
	CMD_SPELLXY           = 14,
	CMD_TSPELLXY          = 15,
	CMD_OPOBJXY           = 16,
	CMD_DISARMXY          = 17,
	CMD_ATTACKID          = 18,
	CMD_ATTACKPID         = 19,
	CMD_RATTACKID         = 20,
	CMD_RATTACKPID        = 21,
	CMD_SPELLID           = 22,
	CMD_SPELLPID          = 23,
	CMD_TSPELLID          = 24,
	CMD_TSPELLPID         = 25,
	CMD_RESURRECT         = 26,
	CMD_OPOBJT            = 27,
	CMD_KNOCKBACK         = 28,
	CMD_TALKXY            = 29,
	CMD_NEWLVL            = 30,
	CMD_WARP              = 31,
	CMD_CHEAT_EXPERIENCE  = 32,
	CMD_CHEAT_SPELL_LEVEL = 33,
	CMD_DEBUG             = 34,
	CMD_SYNCDATA          = 35,
	CMD_MONSTDEATH        = 36,
	CMD_MONSTDAMAGE       = 37,
	CMD_PLRDEAD           = 38,
	CMD_REQUESTGITEM      = 39,
	CMD_REQUESTAGITEM     = 40,
	CMD_GOTOGETITEM       = 41,
	CMD_GOTOAGETITEM      = 42,
	CMD_OPENDOOR          = 43,
	CMD_CLOSEDOOR         = 44,
	CMD_OPERATEOBJ        = 45,
	CMD_PLROPOBJ          = 46,
	CMD_BREAKOBJ          = 47,
	CMD_CHANGEPLRITEMS    = 48,
	CMD_DELPLRITEMS       = 49,
	CMD_PLRDAMAGE         = 50,
	CMD_PLRLEVEL          = 51,
	CMD_DROPITEM          = 52,
	CMD_PLAYER_JOINLEVEL  = 53,
	CMD_SEND_PLRINFO      = 54,
	CMD_SATTACKXY         = 55,
	CMD_ACTIVATEPORTAL    = 56,
	CMD_DEACTIVATEPORTAL  = 57,
	CMD_DLEVEL_0          = 58,
	CMD_DLEVEL_1          = 59,
	CMD_DLEVEL_2          = 60,
	CMD_DLEVEL_3          = 61,
	CMD_DLEVEL_4          = 62,
	CMD_DLEVEL_5          = 63,
	CMD_DLEVEL_6          = 64,
	CMD_DLEVEL_7          = 65,
	CMD_DLEVEL_8          = 66,
	CMD_DLEVEL_9          = 67,
	CMD_DLEVEL_10         = 68,
	CMD_DLEVEL_11         = 69,
	CMD_DLEVEL_12         = 70,
	CMD_DLEVEL_13         = 71,
	CMD_DLEVEL_14         = 72,
	CMD_DLEVEL_15         = 73,
	CMD_DLEVEL_16         = 74,
	CMD_DLEVEL_JUNK       = 75,
	CMD_DLEVEL_END        = 76,
	CMD_HEALOTHER         = 77,
	CMD_STRING            = 78,
	CMD_SETSTR            = 79,
	CMD_SETMAG            = 80,
	CMD_SETDEX            = 81,
	CMD_SETVIT            = 82,
	CMD_RETOWN            = 83,
	CMD_SPELLXYD          = 84,
	CMD_ITEMEXTRA         = 85,
	CMD_SYNCPUTITEM       = 86,
	CMD_KILLGOLEM         = 87,
	CMD_SYNCQUEST         = 88,
	CMD_ENDSHIELD         = 89,
	CMD_AWAKEGOLEM        = 90,
	CMD_NOVA              = 91,
	CMD_SETSHIELD         = 92,
	CMD_REMSHIELD         = 93,
	FAKE_CMD_SETID        = 94,
	FAKE_CMD_DROPID       = 95,
	NUM_CMDS              = 96,
} _cmd_id;
```

Reply:
Will certainly do that but what are current policies for changing types of arguments/struct fields for enums? Currently due to everything being `int` appropriate enum is hard to find in some cases.

Reply:
You kind of just have to know what the code is doing TBH, or trace down the function to see for sure. In this case, `cmd == CMD_CLOSEDOOR` lines up and seems correct.

Reply:
Well but what is the problem with changing type of `cmd` argument to `_cmd_id` if it is expected to be from this enum in all the cases? Is this due to enums being passed as `char`/`unsigned char` ocassionaly?

Reply:
Most of the enums don't have names actually, at least according to the PSX symbol.

Reply:
Yup, many enums were just a bunch of constants. Using the enum as type won't work since the underlying type (`enum name : type { enumerator = constexpr , enumerator = constexpr , ... }`,  the `type` here; e.g. `enum _cmd_id : BYTE { ... }`) cannot be specified before C++11.

Reply:
Well I think it would be helpful to keep them at least as comments in such case